plugins/modules/grafana_folder.py

```diff
@@ -0,0 +1,328 @@
+            description: The date the folder was last updated
+            returned: always
+            type: str
+            sample:
+              - "2018-01-31T17:43:12+01:00"
+        version:
+            description: The folder version
+            returned: always
+            type: int
+            sample:
+              - 1
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url, url_argument_spec, basic_auth_header
+
+__metaclass__ = type
+
+
+class GrafanaFolderInterface(object):
+
+    def __init__(self, module):
+        self._module = module
+        # {{{ Authentication header
+        self.headers = {"Content-Type": "application/json"}
+        if module.params.get('grafana_api_key', None):
+            self.headers["Authorization"] = "Bearer %s" % module.params['grafana_api_key']
+        else:
+            self.headers["Authorization"] = basic_auth_header(module.params['url_username'], module.params['url_password'])
+        # }}}
+        self.grafana_url = module.params.get("url")
+        grafana_version = self.get_version()
+        if grafana_version["major"] < 5:
+            self._module.fail_json(failed=True, msg="Folders API is available starting Grafana v5")
+
+    def _send_request(self, url, data=None, headers=None, method="GET"):
+        if data is not None:
+            data = json.dumps(data, sort_keys=True)
+        if not headers:
+            headers = []
+
+        full_url = "{grafana_url}{path}".format(grafana_url=self.grafana_url, path=url)
+        resp, info = fetch_url(self._module, full_url, data=data, headers=headers, method=method)
+        status_code = info["status"]
+        if status_code == 404:
+            return None
+        elif status_code == 401:
+            self._module.fail_json(failed=True, msg="Unauthorized to perform action '%s' on '%s'" % (method, full_url))
+        elif status_code == 403:
+            self._module.fail_json(failed=True, msg="Permission Denied")
+        elif status_code == 412:
+            error_msg = resp.read()['message']
+            self._module.fail_json(failed=True, msg=error_msg)
+        elif status_code == 200:
+            return self._module.from_json(resp.read())
+        self._module.fail_json(failed=True, msg="Grafana Folders API answered with HTTP %d" % status_code)
+
+    def get_version(self):
+        url = "/api/health"
+        response = self._send_request(url, data=None, headers=self.headers, method="GET")
+        version = response.get("version")
+        major, minor, rev = version.split(".")
+        return {"major": int(major), "minor": int(minor), "rev": int(rev)}
+
+    def create_folder(self, title):
+        url = "/api/folders"
+        folder = dict(title=title)
+        response = self._send_request(url, data=folder, headers=self.headers, method="POST")
+        return response
+
+    def get_folder(self, title):
+        url = "/api/folders?limit=1000"
```

Comment:
Why not use the `/api/search/` url to search for dashboard instead of fetching all folders to find the good one ?

Reply:
Indeed I totally missed the API. I’ll check this

Reply:
Done ! That was an easy change. Thanks for the review !